pkgs/development/tools/build-managers/bazel/default.nix

```diff
@@ -25,21 +27,52 @@ stdenv.mkDerivation rec {
 
   sourceRoot = ".";
 
-  patches = lib.optional enableNixHacks ./nix-hacks.patch;
+  patches = lib.optionals (stdenv.hostPlatform.isDarwin) [ ./darwin.patch ] ++ lib.optionals (enableNixHacks) [ ./nix-hacks.patch ];
 
   # Bazel expects several utils to be available in Bash even without PATH. Hence this hack.
 
-  customBash = writeScriptBin "bash" ''
-    #!${stdenv.shell}
-    PATH="$PATH:${lib.makeBinPath [ coreutils ]}" exec ${bash}/bin/bash "$@"
+  customBash = writeCBin "bash" ''
```

Comment:
This needs to be a native binary instead of a script because otherwise `sandbox-exec` will replace the interpreter with `/bin/sh` instead of using `stdenv.shell`. This might be a security feature, I could repro the behaviour even when using `/bin/bash` as the interpreter, in fact it will use `/bin/sh` as the interpreter even when the first two bytes aren't `#!`, as long as they're not any of the Mach-O byte signatures.

Reply:
Here's a simple repro:

`mybash`:
```
#!/bin/bash

exec /bin/bash "$@"
```

`u.sh`:
```
#!/Users/uri/github/NixOS/nixpkgs/mybash

echo $BASH_VERSION
```

```
$ which bash; bash --version
/Users/uri/homebrew/bin/bash
GNU bash, version 4.4.19(1)-release (x86_64-apple-darwin17.4.0)
Copyright (C) 2016 Free Software Foundation, Inc.
License GPLv3+: GNU GPL version 3 or later <http://gnu.org/licenses/gpl.html>

This is free software; you are free to change and redistribute it.
There is NO WARRANTY, to the extent permitted by law.
$ which sh; sh --version
/bin/sh
GNU bash, version 3.2.57(1)-release (x86_64-apple-darwin17)
Copyright (C) 2007 Free Software Foundation, Inc.

# executing u.sh directly uses mybash as expected
$ ./u.sh
4.4.19(1)-release

# executing u.sh through sandbox-exec uses /bin/sh
$ sandbox-exec -p "(version 1) (allow default)" $PWD/u.sh
3.2.57(1)-release
```

Reply:
See https://github.com/NixOS/nixpkgs/pull/40990#issuecomment-392421415 .

Reply:
Ah yes, scripts are not allowed as a shebang. I wonder if we should generalise this a bit and expose it.